app/validation/validator.py

```diff
@@ -24,23 +23,38 @@ def validate_schema(self, json_to_validate):
 
         errors = []
 
-        errors.extend(self.validate_schema_contains_valid_routing_rules(json_to_validate))
+        numeric_answer_ranges = {}
 
-        errors.extend(self.validate_calculated_ids_in_answers_to_calculate_exists(json_to_validate))
+        for section in json_to_validate['sections']:
+            for group in section['groups']:
 
-        errors.extend(self.validate_routing_rules_has_default_if_not_all_answers_routed(json_to_validate))
+                for block in group['blocks']:
 
-        errors.extend(self.validate_numeric_answer_types(json_to_validate))
+                    if section == json_to_validate['sections'][-1]:
+                        if group == section['groups'][-1]:
+                            if block == group['blocks'][-1]:
+                                errors.extend(self.validate_contains_confirmation_or_summary(block))
 
-        ignored_keys = ['routing_rules', 'skip_conditions']
-        errors.extend(self.validate_duplicates(json_to_validate, ignored_keys, 'id'))
-        errors.extend(self.validate_duplicates(json_to_validate, ignored_keys, 'alias'))
+                    for rule in block.get('routing_rules', []):
+                        errors.extend(self.validate_schema_contains_valid_routing_rules(group['blocks'], 'block', rule))
+                        errors.extend(self.validate_schema_contains_valid_routing_rules(section['groups'], 'group', rule))
+
+                    for question in block.get('questions', []):
+
+                        errors.extend(self.validate_calculated_ids_in_answers_to_calculate_exists(question))
```

Comment:
Small change, and I'm not sure if this is a better way of doing this but what if we check `question['type']` here, instead of in the method for validating. Just a thought

Reply:
I disagree, it spreads the logic for the specific check across 2 functions then.

Reply:
ok can go with that just a thought